requests/request.go

```diff
@@ -218,3 +228,44 @@ func (request *Request) Delete(path string, queryParams map[string]interface{},
 
 	return request.doRequestResponse(req)
 }
+
+func (request *Request) File(path string, params FileUploadParams) (map[string]interface{}, error) {
+	body := &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+
+	// Add the file field to the multipart form data
+	filePart, err := writer.CreateFormFile("file", filepath.Base(params.File.Name()))
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Copy the file contents to the form file part
+	_, err = io.Copy(filePart, params.File)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Add additional form fields
+	for fieldName, fieldValue := range params.Fields {
+		err = writer.WriteField(fieldName, fieldValue)
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	// Close the writer to finalize the form data
+	err = writer.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Create the HTTP request
+	url := fmt.Sprintf("%s%s", request.BaseURL, path)
+
+	req, err := http.NewRequest("POST", url, body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Set the necessary headers
+	req.Header.Add("Content-Type", "multipart/form-data")
```

Comment:
add user agent and extra headers

Reply:
added `request.addRequestHeaders(req, extraHeaders, contentType)` like other http request method